### DISCUSSION TITLE
Compile go to .so

### DISCUSSION THREAD

**Discussion by UnknKriod** [v2fly/v2ray-core#3189] [Q&A]
$ go build -buildmode=c-shared -o libv2ray.so v2ray.go

Here's what the compiler writes to me:

go: downloading github.com/golang/protobuf v1.5.4
go: downloading google.golang.org/protobuf v1.35.1
go: downloading github.com/pires/go-proxyproto v0.8.0
go: downloading golang.org/x/sys v0.26.0
go: downloading github.com/adrg/xdg v0.5.0
-buildmode=c-shared requires exactly one main package

**Comment by UnknKriod** [comment]
I am trying to compile V2Ray for my Android application. I decided to use NDK due to the fact that I do not have the rights to execute binary files in the program folder

Maybe there are ready-made solutions? Tell me, please

**Reply by dyhkwong** [reply]
https://github.com/2dust/AndroidLibV2rayLite

**Comment by UnknKriod** [comment]
Thanks!
